ReactNativeClient/lib/services/KeymapService.ts

```diff
@@ -0,0 +1,284 @@
+		{ accelerator: 'Ctrl+I', command: 'textItalic' },
+		{ accelerator: 'Ctrl+K', command: 'textLink' },
+		{ accelerator: 'Ctrl+`', command: 'textCode' },
+		{ accelerator: 'Ctrl+Shift+T', command: 'insertDateTime' },
+		{ accelerator: 'F6', command: 'focusSearch' },
+		{ accelerator: 'Ctrl+F', command: 'showLocalSearch' },
+		{ accelerator: 'Ctrl+Shift+S', command: 'focusElementSideBar' },
+		{ accelerator: 'Ctrl+Shift+L', command: 'focusElementNoteList' },
+		{ accelerator: 'Ctrl+Shift+N', command: 'focusElementNoteTitle' },
+		{ accelerator: 'Ctrl+Shift+B', command: 'focusElementNoteBody' },
+		{ accelerator: 'F10', command: 'toggleSidebar' },
+		{ accelerator: 'Ctrl+L', command: 'toggleVisiblePanes' },
+		{ accelerator: 'Ctrl+0', command: 'zoomActualSize' },
+		{ accelerator: 'Ctrl+E', command: 'startExternalEditing' },
+		{ accelerator: 'Ctrl+Alt+T', command: 'setTags' },
+		{ accelerator: 'Ctrl+,', command: 'config' },
+		{ accelerator: 'Ctrl+G', command: 'gotoAnything' },
+		{ accelerator: 'F1', command: 'help' },
+	];
+
+	static readonly defaultKeymap = shim.isMac()
+		? KeymapService.defaultKeymap_macOS
+		: KeymapService.defaultKeymap_OtherPlatforms;
+
+	constructor() {
+		super();
+		this.keymap = KeymapService.initKeymap();
+
+		const customKeymapPath = `${Setting.value('profileDir')}/keymap-desktop.json`;
+		if (fs.existsSync(customKeymapPath)) {
+			try {
+				const customKeymap = JSON.parse(fs.readFileSync(customKeymapPath, 'utf-8'));
+				this.setKeymap(customKeymap);
+			} catch (err) {
+				const msg = err.message ? err.message : '';
+				throw new Error(
+					`KeymapService: Failed to override keymap at ${customKeymapPath}!\n\n${msg}`
+				);
+			}
+		}
+	}
+
+	getAccelerator(command: string) {
+		const item = this.keymap[command];
+
+		if (item === undefined) {
+			throw new Error(`KeymapService: Requested "${command}" command is non-existent in the current keymap!`);
+		} else {
+			return item.accelerator;
+		}
+	}
+
+	setAccelerator(command: string, accelerator: string) {
+		// Run through localizeAccelerator() to detect duplicates with high accuracy
+		this.keymap[command].accelerator = KeymapService.localizeAccelerator(accelerator);
+	}
+
+	resetAccelerator(command: string) {
+		const defaultItem = KeymapService.defaultKeymap.find((item => item.command === command));
+		this.setAccelerator(command, defaultItem.accelerator);
+	}
+
+	getKeymap() {
+		const defaultItemsByCommand = KeymapService.initKeymap();
+
+		// Compare each "accelerator" value of the current keymap with the default values
+		// Return a keymap object, which can later be converted to a string and saved to the disk
+		return Object.entries(this.keymap).reduce((customKeymap, [command, { accelerator }]) => {
+			// If the Accelerator has been changed, add into the customKeymap object
+			if (defaultItemsByCommand[command].accelerator !== accelerator) {
+				customKeymap.push({ command, accelerator });
+			}
+			return customKeymap;
+		}, []);
+	}
```

Comment:
This is a get function but it's mutating data. Why does it need to change customKeymap?

Reply:
It's not mutating any data. Initial value of `customKeymap` is just `[]`.

Reply:
This function is not really useful at the moment. It's for finding the changed shortcuts so that it can be saved to `keymap-desktop.json` file.

For example,
1. User changes/resets shortcuts for some commands, and then
2. The service runs this function, convert the `customKeymap` object to JSON and save it.

Reply:
If it's not really used at the moment, would it be possible to remove it? I don't quite understand its purpose and I feel checking for changes, reset, etc. would be a task for the front-end, but maybe I'm missing something, or perhaps the function should be named differently.